[core-kit] app-eselect/* Autogening vi, rust, fontconfig, lib-bin-symlink

org-tekeli-borisp opened the pull request.

org-tekeli-borisp committed:
[core-kit] app-select/eselect-lib-bin-symlink

- autogen app-select/eselect-lib-bin-symlink
- remove app-select/eselect-lib-bin-symlink from gentoo-staging package lists

(cherry picked from commit bd3f17e58a450b31a2a219673be9cb47feb41ec0)
(cherry picked from commit 2cd436e0f9d10a88177ffa77200a42f88275012e)

org-tekeli-borisp committed:
[core-kit] app-select/eselect-fontconfig

- autogen app-select/eselect-fontconfig
- remove app-select/eselect-fontconfig from gentoo-staging package lists

(cherry picked from commit e072612f6f2933e9a20eb9e59b0f2ff48d09b4ec)
(cherry picked from commit 70818a72abbcb8101707108b093ed786888b0320)

org-tekeli-borisp committed:
[core-kit] app-select/eselect-fontconfig template cleanup

(cherry picked from commit 0844ab8f77223559aba85ab1e77d9208e9bc14d0)
(cherry picked from commit 14e63780fd62987b35d542d5250144598e817f06)

org-tekeli-borisp committed:
[core-kit] app-eselect/eselect-rust Migrate eselect-rust to template-based autogeneration

Replaced static ebuild with dynamic template and autogen.yaml for eselect-rust. Simplifies maintenance and enables easier updates by leveraging templated metadata. Removed old manifest and static ebuild in favor of the new approach.

(cherry picked from commit 4bedb14d1e3ddacb3f72154d386ce27ac48d8046)
(cherry picked from commit 8f58159f772c667251e3fa3f6f85b3b905f832a4)

org-tekeli-borisp committed:
[core-kit] app-eselect/eselect-rust Add autotools support to eselect-rust template

Inherited the autotools eclass and added `src_prepare` with `eautoreconf` to enable autotools functionality. This ensures proper preparation of the build environment for projects using this template.

(cherry picked from commit d4a9beeed4f7a202f38e582398d2de8735971c77)

org-tekeli-borisp committed:
[core-kit] app-eselect/eselect-vi: Add template and autogen support

Introduce templated build and autogen configuration for eselect-vi, enabling a more standardized and maintainable setup. This adds metadata, including licensing, description, and dependencies, while structuring the build process for future extensibility.

(cherry picked from commit c247f629c980cd68362e724a570da38603c73b06)
(cherry picked from commit 0ee4b6e9f73f7f4bb256c336dea5b839cf2e4d50)

geaaru commented:
Thanks for all your work. It's really awesome.

I have only a doubt about this. Currently we have a fork of the Portage framework done by Robbins that  I like because permits to use annotation like `PYTHON_COMPAT=( python3+ )`. Are we sure that upgrading Portage tools will be safe if we have this difference. Have you tried if the updated packages work correctly?
Maybe could be better to pin the versions for the same reason?

org-tekeli-borisp commented:
@geaaru  I had metro build failures as the gentoo staging artifacts were not available any more. I autogened these packages as I wanted to have them in my own distfiles repo too.

geaaru commented:
Can you try to add this in your metro build:
```
export GENTOO_MIRRORS="https://distfiles.macaronios.org/mark-distfiles/"
```
It's the new distfiles directory that I'm working and that is updated by mark-devkit.
At the moment the `mark-iii` excluding few packages has all packages available. No issues with eselect stuff if the package is present in `mark-iii`.

org-tekeli-borisp committed:
[core-kit] app-eselect/eselect-vi Update src_install to use vi.eselect instead of vi.eselect.in

This change corrects the installed file in the src_install function by replacing `vi.eselect.in` with `vi.eselect`. It ensures the proper file is deployed to `/usr/share/eselect/modules`. This fix prevents potential issues during installation.

org-tekeli-borisp committed:
[core-kit] app-eselect/eselect-* "Move app-eselect files from 'next' to 'mark' directory

Relocated autogen.yaml and template files for app-eselect components (lib-bin-symlink, vi, fontconfig) from the 'next' directory to the 'mark' directory. This change likely reflects an organizational or naming adjustment for better clarity or workflow alignment."

org-tekeli-borisp commented:
> Have you tried if the updated packages work correctly? Maybe could be better to pin the versions for the same reason?
- rust works
```
~ # eselect rust version
rust.eselect 20200419
eselect 1.4.14
~ # eselect rust list
Available Rust versions:
  [1]   rust-bin-1.85.0 *
```
- fontconfig works
```
~ # eselect fontconfig version
fontconfig.eselect 20091114
eselect 1.4.14
~ # eselect fontconfig list
Available fontconfig .conf files (* is enabled):
  [1]   10-autohint.conf *
  [2]   10-hinting-full.conf *
  [3]   10-hinting-medium.conf *
  [4]   10-hinting-none.conf *
  [5]   10-hinting-slight.conf *
  [6]   10-no-sub-pixel.conf *
  [7]   10-scale-bitmap-fonts.conf *
  [8]   10-sub-pixel-bgr.conf *
  [9]   10-sub-pixel-rgb.conf *
  [10]  10-sub-pixel-vbgr.conf *
  [11]  10-sub-pixel-vrgb.conf *
  [12]  10-unhinted.conf *
  [13]  11-lcdfilter-default.conf *
  [14]  11-lcdfilter-legacy.conf *
  [15]  11-lcdfilter-light.conf *
  [16]  20-unhint-small-dejavu-sans-mono.conf *
  [17]  20-unhint-small-dejavu-sans.conf *
  [18]  20-unhint-small-dejavu-serif.conf *
  [19]  20-unhint-small-vera.conf *
  [20]  25-unhint-nonlatin.conf *
  [21]  30-metric-aliases.conf *
  [22]  40-nonlatin.conf *
  [23]  45-generic.conf *
  [24]  45-latin.conf *
  [25]  49-sansserif.conf *
  [26]  50-user.conf *
  [27]  51-local.conf *
  [28]  57-dejavu-sans-mono.conf *
  [29]  57-dejavu-sans.conf *
  [30]  57-dejavu-serif.conf *
  [31]  60-generic.conf *
  [32]  60-latin.conf *
  [33]  61-urw-bookman.conf *
  [34]  61-urw-c059.conf *
  [35]  61-urw-d050000l.conf *
  [36]  61-urw-fallback-backwards.conf *
  [37]  61-urw-fallback-generics.conf *
  [38]  61-urw-fallback-specifics.conf *
  [39]  61-urw-gothic.conf *
  [40]  61-urw-nimbus-mono-ps.conf *
  [41]  61-urw-nimbus-roman.conf *
  [42]  61-urw-nimbus-sans.conf *
  [43]  61-urw-p052.conf *
  [44]  61-urw-standard-symbols-ps.conf *
  [45]  61-urw-z003.conf *
  [46]  62-croscore-arimo.conf *
  [47]  62-croscore-cousine.conf *
  [48]  62-croscore-symbolneu.conf *
  [49]  62-croscore-tinos.conf *
  [50]  63-source-code-pro.conf *
  [51]  63-source-sans.conf *
  [52]  63-source-serif.conf *
  [53]  65-fonts-persian.conf *
  [54]  65-khmer.conf *
  [55]  65-nonlatin.conf *
  [56]  69-unifont.conf *
  [57]  70-no-bitmaps.conf *
  [58]  70-yes-bitmaps.conf *
  [59]  80-delicious.conf *
  [60]  90-synthetic.conf *
```
- vi first doesn't already but corrected by https://github.com/macaroni-os/kit-fixups/pull/325/commits/5c1e2d74711e786711ad08639fa9b21d8a175246
```
~ # eselect vi version
!!! Error: Can't load module vi
exiting
~ # equery f eselect-vi
 * Searching for eselect-vi ...
 * Contents of app-eselect/eselect-vi-20221122:
/usr
/usr/share
/usr/share/eselect
/usr/share/eselect/modules
/usr/share/eselect/modules/vi.eselect.in
~ # cp /usr/share/eselect/modules/vi.eselect.in /usr/share/eselect/modules/vi.eselect
~ # eselect vi version
vi.eselect @VERSION@
eselect 1.4.14
~ # eselect vi list
Available vi implementations:
  [1]   vim
  [2]   busybox *

```

org-tekeli-borisp commented:
lib-bin-symlink seems to be a eselect lib?
```
 # equery f eselect-lib-bin-symlink
 * Searching for eselect-lib-bin-symlink ...
 * Contents of app-eselect/eselect-lib-bin-symlink-0.1.1:
/usr
/usr/share
/usr/share/eselect
/usr/share/eselect/libs
/usr/share/eselect/libs/bin-symlink.bash
~ # ls /usr/share/eselect/libs/ -al
total 75
drwxr-xr-x 2 root root   14 Mar 23 16:22 .
drwxr-xr-x 4 root root    4 Oct  1  2021 ..
-rw-r--r-- 1 root root 5603 Mar 23 16:22 bin-symlink.bash
-rw-r--r-- 1 root root 3171 Oct 23  2022 config.bash
-rw-r--r-- 1 root root 3418 Oct 23  2022 core.bash
-rw-r--r-- 1 root root 2618 Oct 23  2022 default.eselect
-rw-r--r-- 1 root root 5415 Oct 23  2022 editor-variable.bash
-rw-r--r-- 1 root root 1028 Oct 23  2022 manip.bash
-rw-r--r-- 1 root root 1191 Oct 23  2022 multilib.bash
-rw-r--r-- 1 root root 6684 Oct 23  2022 output.bash
-rw-r--r-- 1 root root 6041 Oct 23  2022 package-manager.bash
-rw-r--r-- 1 root root 2205 Oct 23  2022 path-manipulation.bash
-rw-r--r-- 1 root root 9716 Oct 23  2022 skel.bash
-rw-r--r-- 1 root root 1171 Oct 23  2022 tests.bash

```

geaaru commented:
IIRC lib-bin-synlink contains functions used by other eselect package
